* Allows control of the audio mixer even for media with no audio content
  (e.g. background.avi or tv:// with immediatemode=1)
--- old/mplayer.c	2006-05-27 02:33:57.000000000 +0800
+++ new/mplayer.c	2006-05-27 02:34:09.000000000 +0800
@@ -1567,7 +1567,7 @@
 /// Volume (RW)
 static int mp_property_volume(m_option_t* prop,int action,void* arg) {
 
-    if(!sh_audio) return M_PROPERTY_UNAVAILABLE;
+/*    if(!sh_audio) return M_PROPERTY_UNAVAILABLE;*/
     
     switch(action) {
     case M_PROPERTY_GET:
@@ -1616,7 +1616,7 @@
 /// Mute (RW)
 static int mp_property_mute(m_option_t* prop,int action,void* arg) {
     
-    if(!sh_audio) return M_PROPERTY_UNAVAILABLE;
+/*    if(!sh_audio) return M_PROPERTY_UNAVAILABLE; */
     
     switch(action) {
     case M_PROPERTY_SET:
@@ -3731,6 +3731,20 @@
   mixer.audio_out = audio_out;
   mixer.volstep = volstep;
 }
+else if (!strncmp (filename, "dvd", 3) || strcasestr (filename, ".ifo")) {
+  audio_out=NULL;
+}
+else{
+#if WORDS_BIGENDIAN
+    if(audio_out=init_best_audio_out(audio_driver_list,0,11025,2,AF_FORMAT_S16_BE,0)){
+#else
+    if(audio_out=init_best_audio_out(audio_driver_list,0,11025,2,AF_FORMAT_S16_LE,0)){
+#endif
+        inited_flags|=INITED_AO;
+        mixer.audio_out = audio_out;
+        mixer.volstep = volstep;
+    }
+}
 }
 
 
